packages/orbit-components/src/Drawer/index.tsx

```diff
@@ -40,46 +40,36 @@ const Drawer = ({
   ariaLabel,
 }: Props) => {
   const overlayRef = React.useRef(null);
-  const closeButtonRef = React.useRef<HTMLButtonElement | null>(null);
-  const scrollableRef = React.useRef<HTMLElement | null>(null);
+  const drawerRef = React.useRef<HTMLElement | null>(null);
 
   const [overlayShown, setOverlayShown, setOverlayShownWithTimeout] = useStateWithTimeout<boolean>(
     shown,
     parseFloat(theme.orbit.durationNormal) * 1000,
   );
 
-  const handleOnClose = React.useCallback(
-    (ev: React.MouseEvent<HTMLDivElement>) => {
-      if (onClose && ev.target === overlayRef.current) {
-        onClose();
-      }
-    },
-    [onClose],
-  );
-
-  useFocusTrap(scrollableRef);
-  useLockScrolling(scrollableRef, lockScrolling && overlayShown);
-
-  const handleKeyDown = React.useCallback(
-    (ev: React.KeyboardEvent<HTMLDivElement> | KeyboardEvent) => {
-      if (ev.keyCode === KEY_CODE_MAP.ESC && onClose) {
-        onClose();
-      }
-    },
-    [onClose],
-  );
+  useFocusTrap(drawerRef);
+  useLockScrolling(drawerRef, lockScrolling && overlayShown);
 
   React.useEffect(() => {
-    closeButtonRef.current?.focus();
-
     if (overlayShown !== shown) {
       if (shown) {
         setOverlayShown(true);
       } else if (!shown) {
         setOverlayShownWithTimeout(false);
       }
     }
-  }, [overlayShown, setOverlayShown, shown, setOverlayShownWithTimeout, onClose]);
+  }, [overlayShown, setOverlayShown, shown, setOverlayShownWithTimeout]);
+
+  React.useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === "Escape" && onClose) {
```

Comment:
what about adding a condition if the drawer is opened/closed? Now, I can press the ESC button and call the function whenever I want although the drawer is not shown.